### PR TITLE
New Colour and Typography generators and helpers

### PR DESCRIPTION
**Pull request by jsundt**
**Purpose:**
Move text and colour classes to new naming convention. Set better defaults for each of them. More utilities for easier prototyping and design testing.

**For dissucion:** 
I'm still not quite convinced by the naming of Note, label and caption, or if all of them are even necessary. Maybe the label and caption style could be a component of the size rather than a object?

**Changes:**
– Type inherit colour by default
– Type hierarchy is no longer random, and have more consistent naming
– Added new background colour utilities
– Added new text colour utilities
– Added new text opacity utilities
– Added new colour by state utilities

### CODE REVIEW

File: vendor/assets/stylesheets/fenrir/configs/defaults.scss [hunk 1]
```diff
@@ -194,16 +203,37 @@ $title-font: $charlie-font !default;
...
+$type-sizes: (
+  'heading-1': (87px, 100px),
+  'heading-2': (65px, 75px),
+  'heading-3': (49px, 56px),
+  'heading-4': (37px, 42px),
+  'heading-5': (28px, 32px),
+  'heading-6': (21px, 24px),
+  'body':      (16px, 18px),
+  // 'label':     (12px, 14px),
+  // 'note':      (10px, 12px),
```
Comment (grannellmat) [inline vendor/assets/stylesheets/fenrir/configs/defaults.scss]: Should we remove these guys if no longer necessary?

**Comment by grannellmat** [comment]
Really nice stuff. 🇳🇴 

I like the idea of caption and labels being components. Structurally they are identical to notes, but stylistic they have differences. If we follow our general object rule that would assume they become components?

**Comment by jsundt** [comment]
That could potentially simplify the generator a bit. I guess the other thing we'd need to consider is how we'd want to include that in other components. Usually I guess we only include objects in components rather than components in components. 

But that said, if we need component specific text we could build it out from the base note size again instead of overriding label or caption. And they'll just stay as unmutable but reusable text classes.

**Review by grannellmat — CHANGES_REQUESTED** [review]
👍

**Review by grannellmat — APPROVED** [review]

File: vendor/assets/stylesheets/fenrir/configs/defaults.scss (hunk 1 above)
Comment (jsundt) [reply vendor/assets/stylesheets/fenrir/configs/defaults.scss]: Yeah might as well

**Pull request merged**